src/rulesets/Base/nondiff.jl

```diff
@@ -54,15 +54,14 @@
 @non_differentiable invperm(::AbstractArray{Bool})
 @non_differentiable invpermute!(::Any, ::AbstractArray{Bool})
 @non_differentiable mapslices(::Any, ::AbstractArray{Bool})
-@non_differentiable Matrix(::AbstractArray{Bool})
```

Comment:
this is actually taken care of in https://github.com/JuliaDiff/ChainRules.jl/blob/main/src/rulesets/Base/array.jl#L15-L19

and the project to makes sure it is always NoTangent() for arrays of booleans, and it compiles away (at least the time is the same as with the dedicated rule)